proof/refine/ARM_HYP/Arch_R.thy

```diff
@@ -1306,6 +1306,24 @@ lemma archThreadSet_VCPU_Some_corres[corres]:
   apply (simp add: arch_tcb_relation_def)
   done
 
+crunches dissociateVCPUTCB
+  for no_0_obj'[wp]: no_0_obj'
+  and ksCurThread[wp]: "\<lambda>s. P (ksCurThread s)"
+  (simp: crunch_simps wp: crunch_wps)
+
+lemma vcpuSwitch_corres'':
+  assumes "vcpu' = vcpu"
+  shows
+  "corres dc (\<lambda>s. (vcpu \<noteq> None \<longrightarrow> vcpu_at  (the vcpu) s) \<and> valid_arch_state s)
```

Comment:
strange indent (at same level as `shows`) and not really needed assumes/shows... is this only moved?

Reply:
This is very similar to `vcpuSwitch_corres'`, and I copied that lemma and modified it here to much more easily apply with the `corressimp` in the `associateVCPUTCB_corres` proof. I will at least remove the `assumes/shows`